clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -1704,6 +1704,20 @@ bool SemaHLSL::CheckBuiltinFunctionCall(unsigned BuiltinID, CallExpr *TheCall) {
       return true;
     break;
   }
+  case Builtin::BI__builtin_hlsl_cross: {
+    if (SemaRef.checkArgCount(TheCall, 2))
+      return true;
+    if (CheckVectorElementCallArgs(&SemaRef, TheCall))
+      return true;
+    if (CheckFloatOrHalfRepresentations(&SemaRef, TheCall))
+      return true;
+
```

Comment:
Maybe we should do a  vector size check to prevent misuse of the builtin? Downsize is the check isn't triggerable from the hlsl function name.

Reply:
There are also other asserts to catch when the size isn't 3, I don't think this is needed.

Reply:
If thats the case could you add a test to `clang/test/CodeGenHLSL/builtins/cross.hlsl` As long as this gets caugh by an error then we should be good.

```hlsl
float2 builtin_cross_float2(float2  p1, float2  p2)
{
  return __builtin_hlsl_cross(p1, p2);
}
```

Reply:
I added a diagnostic and put the test in cross-errors.hlsl instead.